src/sentry/api/endpoints/oauth_userinfo.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework.authentication import get_authorization_header
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.models import ApiToken, UserEmail
+
+
+@control_silo_endpoint
+class OAuthUserInfoEndpoint(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    def get(self, request: Request) -> Response:
+        try:
+            access_token = get_authorization_header(request).split()[1].decode("utf-8")
+        except IndexError:
+            return Response("No access token found.", status=401)
+        try:
+            token_details = ApiToken.objects.get(token=access_token)
+        except ApiToken.DoesNotExist:
+            return Response(status=400)
+
+        scopes = token_details.get_scopes()
+        if "openid" not in scopes:
+            return Response(status=403)
```

Comment:
An error message would be good here too.

Reply:
Fixed